src/arch/riscv/kernel/boot.c

```diff
@@ -214,11 +214,13 @@ extern char trap_entry[];
 BOOT_CODE static void
 init_cpu(void)
 {
-
     /* Write trap entry address to stvec */
-    write_csr(stvec, trap_entry);
+    write_csr_env(tvec, trap_entry);
```

Comment:
You had a previous commit that claimed "Replace S-mode instructions/registers with privilege-aware definitions", seems like this should have been part of that?